monai/transforms/spatial/array.py

```diff
@@ -2091,15 +2018,25 @@ def __init__(
             scale_params=scale_params,
             affine=affine,
             dtype=dtype,
+            align_corners=align_corners,
             device=device,
         )
         self.image_only = image_only
         self.norm_coord = not normalized
-        self.resampler = Resample(norm_coords=self.norm_coord, device=device, dtype=dtype)
+        self.resampler = Resample(norm_coords=self.norm_coord, device=device, dtype=dtype, align_corners=align_corners)
         self.spatial_size = spatial_size
         self.mode = mode
         self.padding_mode: str = padding_mode
 
+        self._grid = None
+        self._affine = None
+        self._sp_size = None
```

Comment:
I didn't find where we use these 3 memeber vars, and maybe it's not thread-safe?
CC @wyli .

Thanks.

Reply:
thanks, these are not needed any more, I'll remove